cpp/src/strings/strings_column_view.cpp

```diff
@@ -42,26 +45,21 @@ strings_column_view::offset_iterator strings_column_view::offsets_end() const
   return offsets_begin() + size() + 1;
 }
 
-column_view strings_column_view::chars() const
-{
-  CUDF_EXPECTS(num_children() > 0, "strings column has no children");
-  return child(chars_column_index);
-}
-
-size_type strings_column_view::chars_size() const noexcept
+size_type strings_column_view::chars_size(rmm::cuda_stream_view stream) const noexcept
 {
   if (size() == 0) return 0;
-  return chars().size();
+  return detail::get_value<size_type>(offsets(), offset() + size(), stream);
```

Comment:
This function seems to be called a lot. Can we cache it into a member variable?

Reply:
Yes. caching is required. That change is moved to next PR to not overload the work of reviewers of this PR.